src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -0,0 +1,631 @@
+        {
+            _runId = _reader.ReadElementContentAsString();
+        }
+
+        private void ParseCreatedTimestamp()
+        {
+            string date = _reader.GetAttribute(_strings.DateAttribute);
+            string time = _reader.GetAttribute(_strings.TimeAttribute);
+            if (!string.IsNullOrEmpty(date) && !string.IsNullOrEmpty(time))
+            {
+                string dateTime = date + "T" + time;
+                _invocation.StartTime = DateTime.Parse(dateTime, CultureInfo.InvariantCulture);
+            }
+
+            // Step past the empty element.
+            _reader.Read();
+        }
+
+        private void ParseBuild()
+        {
+            _reader.Read();
+            while (!AtEndOf(_strings.Build))
+            {
+                if (AtStartOfNonEmpty(_strings.BuildId))
+                {
+                    _automationId = _reader.ReadElementContentAsString();
+                }
+                else if (AtStartOfNonEmpty(_strings.SourceFiles))
+                {
+                    ParseSourceFiles();
+                }
+                else
+                {
+                    _reader.Read();
+                }
+            }
+        }
+
+        private void ParseSourceFiles()
+        {
+            _reader.Read();
+            while (!AtEndOf(_strings.SourceFiles))
+            {
+                if (AtStartOfNonEmpty(_strings.File))
+                {
+                    ParseFile();
+                }
+                else
+                {
+                    _reader.Read();
+                }
+            }
+        }
+
+        private void ParseFile()
+        {
+            string fileType = _reader.GetAttribute(_strings.TypeAttribute);
+            int length = 0;
+            string sizeAttribute = _reader.GetAttribute(_strings.SizeAttribute);
+            if (sizeAttribute != null)
+            {
+                if (!int.TryParse(sizeAttribute, out length))
+                {
+                    length = 0;
+                }
+            }
+
+            string fileName = null;
+            _reader.Read();
+            while (!AtEndOf(_strings.File))
+            {
+                if (AtStartOfNonEmpty(_strings.Name))
+                {
+                    fileName = _reader.ReadElementContentAsString();
+                }
+                else
+                {
+                    _reader.Read();
+                }
+            }
+
+            if (!string.IsNullOrEmpty(fileName))
+            {
+                _fileDictionary.Add(
+                    fileName,
+                    new FileData
+                    {
+                        MimeType = FprTypeToMimeType(fileType),
+                        Length = length
+                    });
+            }
+        }
+
+        private string FprTypeToMimeType(string fileType)
+        {
+            switch (fileType)
+            {
+                case "xml":
+                    return "text/xml";
```

Comment:
>text/xml [](start = 28, length = 8)

Can't we use the MimeType helper to get these values? may need to add tsql. The MimeType helper (in \Writers\) should accept the fileName directly. Note that this helper assumes a binary mime type when encountering something unknown.